addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,428 @@
+        token = self._l10n_eg_get_einvoice_token(invoice)
+        api_domain = self._l10n_eg_get_eta_api_domain()
+        if isinstance(token, dict) and token.get('error', False):
+            return {invoice: {
+                'error': token.get('error', False),
+                'blocking_level': token.get('blocking_level', False)
+            }
+            }
+        request_url = '%s/api/v1.0/documentsubmissions' % api_domain
+
+        request_payload = {
+            'documents': [invoice_json]
+        }
+
+        data = json.dumps(request_payload, ensure_ascii=False, indent=4).encode('utf-8')
+
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': 'Bearer %s' % token}
+        try:
+            request_response = requests.post(
+                request_url, data=data, headers=headers, timeout=(5, 10))
+            submission_data = request_response.json()
+        except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.MissingSchema, requests.exceptions.Timeout, requests.exceptions.HTTPError) as ex:
+            return {invoice: {'error': str(error), 'blocking_level': 'warning'}}
+
+        if submission_data:
+            if submission_data.get('error'):
+                return {invoice: {
+                    'error': submission_data.get('error'),
+                    'blocking_level': 'error'
+                }}
+            if submission_data.get('rejectedDocuments', False) and isinstance(
+                    submission_data.get('rejectedDocuments'), list):
+                return {invoice: {
+                    'error': str(submission_data.get('rejectedDocuments')[0].get('error')),
+                    'blocking_level': 'error'
+                }}
+            if (
+                submission_data.get('submissionId')
+                and submission_data.get('submissionId') is not None
+                and submission_data.get('acceptedDocuments')
+            ):
+                return self._extracted_from__l10n_eg_edi_post_invoice_web_service_37(
+                    submission_data, invoice
+                )
+
+            return {invoice: {
+                'success': False,
+            }}
+
+    # TODO Rename this here and in `_l10n_eg_edi_post_invoice_web_service`
+    def _extracted_from__l10n_eg_edi_post_invoice_web_service_37(self, submission_data, invoice):
+        invoice.l10n_eg_submission_id = submission_data.get('submissionId')
+        uuid = submission_data.get('acceptedDocuments')[0].get('uuid')
+        invoice.l10n_eg_uuid = uuid
+        invoice.l10n_eg_long_id = submission_data.get(
+            'acceptedDocuments')[0].get('longId')
+        invoice.l10n_eg_internal_id = submission_data.get('acceptedDocuments')[
+            0].get('internalId')
+        invoice.l10n_eg_hash_key = submission_data.get('acceptedDocuments')[
+            0].get('hashKey')
+        return {invoice: {
```

Comment:
It might be better to make these fields computed from the attachment.  At least, if we put the response there as well.  It would save a lot of fields on the account.move